### ISSUE TITLE
Auth.SignIn() success with currentAuthenticatedUser() but currentCredentials() remains un-authenticated.

### ISSUE THREAD

**Issue by QuantJames** [aws-amplify/amplify-js#4242]
I am using Amplify Auth to authenticate users in a react-native application using a cognito user pool. This works fine, and `currentAuthenticatedUser()` returns the correct user data and valid JWT tokens for making GraphQL queries, _however_, `currentCredentials()` is remaining in an un-authenticated state, making it impossible to access services like S3 within my application.

According to the documentation [here](https://aws-amplify.github.io/docs/js/authentication) (under 'AWS Auth'), calling `SignIn()` should result in requests automatically being signed. Is this a bug or am I missing something? This is a major blocker on my current project now.

Many thanks.

** Which Category is your question related to? **
Auth
** What AWS Services are you utilizing? **
Cognito User Pools, Identity Pools

**Comment by QuantJames** [comment]
bump

**Comment by keithkaikai** [comment]
I am having the same issue

**Comment by Amplifiyer** [comment]
Hi @keithkaikai  and @QuantJames, I tried reproducing the issue but I was not able to. Here are the steps I followed

1) Use WithAuthenticator HOC to handle signIn OR Auth.signIn() with userpool username and password
2) Check Auth.currentAuthenticatedUser()
3) Check Auth.currentCredentials()

In both cases (using HOC or API call) I am getting successful results for both currentAuthenticatedUser and currentCredentials.

I've tested this with aws-amplify/auth@2.2.1 and aws-amplify-react-native@2.2.3

Will you be able to provide a sample app where we can reproduce this?
